### ISSUE TITLE
Support for copy/paste

### ISSUE THREAD

**Issue by Leon99** [dotnet/winforms#10055]
### Environment

N/A

### .NET version

.NET 6

### Did this work in a previous version of Visual Studio and/or previous .NET release?

.NET Framework

### Issue description

According to https://github.com/dotnet/winforms/issues/1831, there were changes in .NET 6 that broke the copy/paste support. However, I couldn't find anything like "copy/paste is no longer supported in .NET Core" in the official documentation. Is there any other recommended way to implement it, or it's an omission from the documentation and there won't be any support for copy/paste now?

### Steps to reproduce

N/A

**Comment by elachlan** [comment]
.NET deprecated the binary formatter due to fundamental security issues with it. This among other things has lead the team to refactor the winforms interop layer to add better support for COM Wrappers to enable better copy paste support. It's still ongoing and it looks like better support will be added in .NET9.

See #6269

**Comment by elachlan** [comment]
Cc: @JeremyKuhne

**Comment by lonitra** [comment]
@Leon99 Could you clarify your scenario here? Is copy and paste not working for you in the designer? If BinaryFormatter is enabled, copy and paste should be working

**Comment by Leon99** [comment]
> copy and paste not working for you in the designer

Correct

> Is copy and paste not working for you in the designer? If BinaryFormatter is enabled, copy and paste should be working

https://github.com/dotnet/winforms/issues/1831 explains the reason why it doesn't work.

**Comment by JeremyKuhne** [comment]
@Leon99 to be super explicit here, is `CodeDomSerializationStore` the one you are referring to? If you can give us a detailed repro we will be better able to give you guidance.

Copy/paste still work in general, we removed `[Serializable]` from a few types as part of the greater effort to reduce exposure to security issues with arbitrary object binary serialization.

**Comment by Leon99** [comment]
@JeremyKuhne, yes, `CodeDomSerializationStore` is the type responsible for the serialization/deserialization of controls, which is essential to copy/paste. I'm not sure what you mean by "Copy/paste still work in general", given that it requires serialization/deserialization of controls to get the data to be placed on the clipboard, which doesn't work anymore.

We solved the issue, but it took lots of effort as it involved switching to another serializer that doesn't require `[Serializable]` but supports all the involved types, and modification of a whole bunch of WinForms types, as most of the involved infrastructure is _internal_.

**Comment by elachlan** [comment]
Also related: #9110

**Comment by JeremyKuhne** [comment]
> I'm not sure what you mean by "Copy/paste still work in general"

@Leon99 Copy and paste still works for standard scenarios (say, string). In order for us to follow up on your specific scenario we need an explicit repro (code) to fully understand our options here and to give you clear answers. You've also stated that you have a workaround, is this actually a blocker?

@Tanya-Solyanik had marked this type as being only used [for internal scenarios](https://github.com/dotnet/winforms/issues/889#issuecomment-513040581) but clearly this isn't true.

I'm not specifically opposed to putting the attribute back, but:

1. We would need to have clear details on what is blocked so we determine where this work falls in the backlog
2. As mentioned, the `BinaryFormatter` is depreciated and will likely be disabled by default for WinForms projects in .NET 9 (it already is disabled by default for other .NET 8 projects)

**Comment by JeremyKuhne** [comment]
> Also related: #9110

We're unlikely to make this particular type work without the `BinaryFormatter`. It appears to not have constraints on the data that is being serialized. #9110 is for adding handling for types that we know the full closure of.

**Comment by Leon99** [comment]
> We would need to have clear details on what is blocked

Copy/paste of controls in the WinForms designer.

> You've also stated that you have a workaround, is this actually a blocker?

We're not blocked anymore, but the workaround is not particularly maintainable.
Besides, given that it's clearly a functional regression, and the workaround involves messing with many internal to base .NET library types, would be nice to see something like "copy/paste of controls in WinForms designer is no longer officially supported in .NET Core" in the documentation so people considering migration know what to expect.

**Comment by RussKie** [comment]
https://github.com/dotnet/announcements/issues/284
